File: TGAPEngine/src/Light.cpp
#include "Light.h"

namespace engine{
	Light::Light(vec4 direction, vec3 color, vec4 location, bool isDirectional) :direction(direction), color(color), location(location), m_isDirectional(isDirectional)
	{
		//ctor
	}
	Light::Light(float xdir, float ydir, float zdir, float r, float g, float b, float x, float y, float z, bool isDirectional)
	{
		this->direction = vec4(xdir, ydir, zdir, 0.0f);
		this->color = vec3(r, g, b);
		this->location = vec4(x, y, z, 1.0f);
		this->m_isDirectional = isDirectional;

	}
	Light::~Light()
	{
		//dtor
	}


}

File: CS/TGAPE_CS/Engine/Renderer/Light.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;

namespace TGAPE_CS.Engine.Renderer
{//TODO:make light an entity so it can be used in object tree if possible if not change to a struct
    public class Light
    {
        public bool IsDirectional { get; set; }
        public Vector4 Direction { get; set; }
        public Vector3 Color { get; set; }
        public Vector4 Location { get; set; }
        public Light(Vector4 direction, Vector3 color, Vector4 location, bool isDirectional)
        {
            Direction = direction;
            Color = color;
            Location = location;
            IsDirectional = isDirectional;
        }
    }
}
